insights/client/apps/compliance/__init__.py

```diff
@@ -0,0 +1,66 @@
+from insights.client.connection import InsightsConnection
+from insights.util.canonical_facts import get_canonical_facts
+from subprocess import Popen, PIPE, STDOUT
+from sys import exit
+from re import findall
+from platform import linux_distribution
+from glob import glob
+from logging import getLogger
+
+OSCAP_RESULTS_OUTPUT = '/tmp/oscap_results.xml'
+NONCOMPLIANT_STATUS = 2
+COMPLIANCE_CONTENT_TYPE = 'application/vnd.redhat.compliance.something+tgz'
+POLICY_FILE_LOCATION = '/usr/share/xml/scap/ssg/content/'
+logger = getLogger(__name__)
+
+
+class ComplianceClient:
+    def __init__(self, config):
+        self.config = config
+        self.conn = InsightsConnection(config)
+        self.hostname = get_canonical_facts().get('fqdn', '')
+
+    def oscap_scan(self):
+        policies = self.get_policies()
+        if not policies:
+            logger.error("ERROR: System does not exist!\n")
+            exit(1)
```

Comment:
The way the client / core wrapper works, to exit the process properly, you should use `sig_kill_bad` (failure exit code) or `sig_kill_ok` (success exit code) from the `constants.py` file. If you exit with 1 it sees that as a "bad egg" i.e., a traceback happened, so it will try to run again with the next available egg.